flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobServerConnection.java

```diff
@@ -540,7 +526,7 @@ else if (type == NAME_ADDRESSABLE) {
 					// we should make the local and remote file deletion atomic, otherwise we might risk not
 					// removing the remote file in case of a concurrent put operation
 					if (blobFile.exists() && !blobFile.delete()) {
```

Comment:
you're right, this could be improved as well - the next PR in this series is removing this code though